<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>homepage</class>
 <widget class="QWidget" name="homepage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Iosevka</family>
   </font>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QWidget" name="verticalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>280</x>
     <y>360</y>
     <width>160</width>
     <height>146</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <layout class="QVBoxLayout" name="Blur">
    <item>
     <widget class="QPushButton" name="MeanBlur">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Mean Blur</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="BoxFilter">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Box Filter</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="GaussianBlur">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Gaussian Blur</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="MedianBlur">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Median Blur</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="Hist">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>70</y>
     <width>256</width>
     <height>200</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <property name="text">
    <string>             Hist</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>330</y>
     <width>91</width>
     <height>24</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <property name="text">
    <string>ChooseImage</string>
   </property>
   <property name="autoRepeat">
    <bool>false</bool>
   </property>
   <property name="default">
    <bool>false</bool>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QListView" name="UserlistView">
   <property name="geometry">
    <rect>
     <x>460</x>
     <y>360</y>
     <width>141</width>
     <height>221</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <property name="movement">
    <enum>QListView::Movement::Static</enum>
   </property>
  </widget>
  <widget class="QLCDNumber" name="SaltNoiseLCDNumber">
   <property name="geometry">
    <rect>
     <x>190</x>
     <y>410</y>
     <width>64</width>
     <height>23</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
   <property name="smallDecimalPoint">
    <bool>true</bool>
   </property>
   <property name="segmentStyle">
    <enum>QLCDNumber::SegmentStyle::Flat</enum>
   </property>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>510</y>
     <width>431</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <layout class="QHBoxLayout" name="BrightnesLayout">
    <item>
     <widget class="QSlider" name="Brightnes">
      <property name="minimumSize">
       <size>
        <width>340</width>
        <height>16</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>340</width>
        <height>16</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
      <property name="sliderPosition">
       <number>50</number>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="tickInterval">
       <number>0</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="BrightnessText">
      <property name="minimumSize">
       <size>
        <width>80</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Brightness</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="ImagePass">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>330</y>
     <width>611</width>
     <height>21</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <property name="text">
    <string>None</string>
   </property>
  </widget>
  <widget class="QLabel" name="HistoryText">
   <property name="geometry">
    <rect>
     <x>610</x>
     <y>360</y>
     <width>54</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <property name="text">
    <string>History</string>
   </property>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_5">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>460</y>
     <width>160</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <layout class="QHBoxLayout" name="KernelSizeLayout">
    <item>
     <widget class="QLabel" name="KernelSizeText">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>KernelSize</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QSpinBox" name="KernelSize">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>9</number>
      </property>
      <property name="singleStep">
       <number>2</number>
      </property>
      <property name="value">
       <number>5</number>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>290</y>
     <width>351</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <layout class="QHBoxLayout" name="SaveExport">
    <item>
     <widget class="QPushButton" name="SaveBuf">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>SaveBuf</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="Cancel">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Cancel</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="Export">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Export</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="ExportHist">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>ExportHist</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_3">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>550</y>
     <width>431</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <layout class="QHBoxLayout" name="ContrastLayout">
    <item>
     <widget class="QSlider" name="Contrast">
      <property name="minimumSize">
       <size>
        <width>340</width>
        <height>16</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>340</width>
        <height>16</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
      <property name="sliderPosition">
       <number>50</number>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="tickInterval">
       <number>0</number>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="ContrastText">
      <property name="minimumSize">
       <size>
        <width>80</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Contrast</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_4">
   <property name="geometry">
    <rect>
     <x>330</x>
     <y>20</y>
     <width>426</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <layout class="QHBoxLayout" name="GrayNeed">
    <item>
     <widget class="QPushButton" name="ConvertGary">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>ConvertGray</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="Threshold">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Threshold</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="Erode">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Erode</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="Dilate">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Dilate</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="PictureArea">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>20</y>
     <width>300</width>
     <height>300</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <property name="frameShape">
    <enum>QFrame::Shape::NoFrame</enum>
   </property>
   <property name="text">
    <string>    PicturePreview</string>
   </property>
  </widget>
  <widget class="QLCDNumber" name="GaussionNoiseLCDNumber">
   <property name="geometry">
    <rect>
     <x>190</x>
     <y>370</y>
     <width>64</width>
     <height>23</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
   <property name="smallDecimalPoint">
    <bool>true</bool>
   </property>
   <property name="mode">
    <enum>QLCDNumber::Mode::Dec</enum>
   </property>
   <property name="segmentStyle">
    <enum>QLCDNumber::SegmentStyle::Flat</enum>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>360</y>
     <width>160</width>
     <height>86</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Iosevka</family>
    </font>
   </property>
   <layout class="QVBoxLayout" name="Noise">
    <item>
     <widget class="QPushButton" name="GaussNoise">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Gauss Noise</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="SaltNoise">
      <property name="font">
       <font>
        <family>Iosevka</family>
       </font>
      </property>
      <property name="text">
       <string>Salt Noise</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
